cli/azd/pkg/azdo/project.go

```diff
@@ -0,0 +1,190 @@
+			Capabilities: &capabilities,
+		},
+	}
+	res, err := coreClient.QueueCreateProject(ctx, args)
+	if err != nil {
+		return nil, err
+	}
+
+	operationsClient := operations.NewClient(ctx, connection)
+
+	getOperationsArgs := operations.GetOperationArgs{
+		OperationId: res.Id,
+	}
+
+	projectCreated := false
+	maxCheck := 10
+	count := 0
+
+	for !projectCreated {
+		operation, err := operationsClient.GetOperation(ctx, getOperationsArgs)
+		if err != nil {
+			return nil, err
+		}
+
+		if *operation.Status == "succeeded" {
+			projectCreated = true
+		}
+
+		if count >= maxCheck {
+			return nil, fmt.Errorf("error creating azure devops project %s", name)
+		}
+
+		count++
+		time.Sleep(800 * time.Millisecond)
+	}
+
+	project, err := getProjectByName(ctx, connection, name)
+	if err != nil {
+		return nil, err
+	}
+	return project, nil
+}
+
+// prompts the user for a new AzDo project name and creates the project
+// returns project name, project id, error
+func GetProjectFromNew(ctx context.Context, repoPath string, connection *azuredevops.Connection, env *environment.Environment, console input.Console) (string, string, error) {
+	var project *core.TeamProjectReference
+	currentFolderName := filepath.Base(repoPath)
+	var projectDescription string = AzDoProjectDescription
+
+	for {
+		name, err := console.Prompt(ctx, input.ConsoleOptions{
+			Message:      "Enter the name for your new Azure DevOps Project OR Hit enter to use this name:",
+			DefaultValue: currentFolderName,
+		})
+		if err != nil {
+			return "", "", fmt.Errorf("asking for new project name: %w", err)
+		}
+		var message string = ""
+		newProject, err := createProject(ctx, connection, name, projectDescription, console)
+		if err != nil {
+			message = err.Error()
+		}
+		if strings.Contains(message, fmt.Sprintf("The following project already exists on the Azure DevOps Server: %s", name)) {
+			console.Message(ctx, fmt.Sprintf("error: the project name '%s' is already in use\n", name))
+			continue // try again
+		} else if strings.Contains(message, "The following name is not valid") {
+			console.Message(ctx, fmt.Sprintf("error: the project name '%s' is not a valid Azure DevOps project Name. See https://aka.ms/azure-dev/azdo-project-naming\n", name))
+			continue // try again
+		} else if err != nil {
+			return "", "", fmt.Errorf("creating project: %w", err)
+		} else {
+			project = newProject
+			break
+		}
+	}
+
+	return *project.Name, project.Id.String(), nil
+}
+
+// return an azdo project by name
+func getProjectByName(ctx context.Context, connection *azuredevops.Connection, name string) (*core.TeamProjectReference, error) {
+	coreClient, err := core.NewClient(ctx, connection)
+	if err != nil {
+		return nil, err
+	}
+
+	args := core.GetProjectsArgs{}
+	getProjectsResponse, err := coreClient.GetProjects(ctx, args)
+	if err != nil {
+		return nil, err
+	}
+
+	projects := getProjectsResponse.Value
+	for _, project := range projects {
+		if *project.Name == name {
```

Comment:
Just want to confirm that we want to do a case sensitive match against the project name that they enter.

Reply:
Yes, that is correct. The API actually normalizes the casing when creating a project to check for duplicates. However, this line specifically checks the newly created project against the project name supplied by the user. It's used because the create operation doesn't return a project Id. Tested it many and it works as expected, even if you enter a weird case for an existing project